File: Chapter 05/JS/section_5_4_server_monitor/after/tests/fakeAlertPort.js
const AlertPort = require('../core/ports/alertPort');

/**
 * Fake port for isolated testing.
 * Extracted into its own file so it can be reused across multiple test suites.
 */
class FakeAlertPort extends AlertPort {
    constructor() {
        super();
        this.sentMessages = [];
    }

    sendAlert(message) {
        this.sentMessages.push(message);
    }
}

module.exports = FakeAlertPort;

File: Chapter 05/Python/section_5_4_server_monitor/after/tests/fake_alert_port.py
from ..core.ports.alert_port import AlertPort

class FakeAlertPort(AlertPort):
    """
    Fake port for isolated testing.
    Extracted into its own file so it can be reused across multiple test suites.
    """

    def __init__(self):
        self.sent_messages = []

    def send_alert(self, message: str) -> None:
        self.sent_messages.append(message)